gcloud-java-dns/README.md

```diff
@@ -0,0 +1,324 @@
+for(DnsRecord record : records) {
+  System.out.println(record)
+}
+```
+
+You can now instruct your domain registrar to [update the domain name servers] (https://cloud.google.com/dns/update-name-servers)
+to these. As soon as this happens and the change propagates through cached values in DNS resolvers,
+all the DNS queries will be directed and answered by the Google Cloud DNS service.
+
+#### Creating DNS Records
+Now that we have an zone, we should add some DNS records. The DNS records held within zones are
+modified by so-called "change requests". In this example, we create and apply a change request to
+our zone that creates a DNS record of type A and points URL www.someexampledomain.com to
+IP address 12.13.14.15. Start by adding `import com.google.gcloud.dns.ChangeRequest` and proceed
+with:
+
+```java
+// Prepare a www.someexampledomain.com. type A record with ttl of 24 hours
+String ip = "12.13.14.15";
+DnsRecord toCreate = DnsRecord.builder("www.someexampledomain.com.", DnsRecord.Type.A)
+  .ttl(24, TimeUnit.HOURS)
+  .addRecord(ip)
+  .build();
+// Make a change
+ChangeRequest.Builder changeBuilder = ChangeRequest.builder().add(toCreate);
+// Build and apply the change request to our zone
+ChangeRequest changeRequest = zone.applyChangeRequest(changeBuilder.build());
+```
+
+The `addRecord` (and also `records`) function of `DnsRecord.Builder` accepts records in the form of
+strings (and a list of strings respectively). The format of the strings (and the number of strings
+in the list respectively) depends on the type of the DNS record to be added. More information on the
+supported DNS record types and record formats can be found [here](https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types).
+
+When the change request is applied, it is registered with the Cloud DNS service for processing. We
+can wait for its completion as follows:
+
+```java
+while (ChangeRequest.Status.PENDING.equals(changeRequest.status())) {
+  Thread.sleep(500L);
+  changeRequest = dns.getChangeRequest(zone.name(), changeRequest.id());
+}
+System.out.println("The change request has been applied.");
+```
+
+Change requests are applied atomically to all the assigned DNS servers at once. Note that when this
+happens, it may still take a while for the change to be registered by the DNS cache resolvers in the
+Internet. See more on this topic [here](https://cloud.google.com/dns/monitoring).
+
+#### Listing Zones and DNS Records
+Suppose that you've added more zones and DNS records in them, and now you want to see what they are.
+First, import the following class for easy iteration:
+
+```java
+import java.util.Iterator;
+```
+
+Then add the following code to list all your zones and DNS records.
+
+```java
+// List all your zones
+Iterator<Zone> projectZones = dns.listZones().iterateAll();
+System.out.println("My zones:");
+while (projectZones.hasNext()) {
+  System.out.println(projectZones.next());
+}
+
+// List the DNS records in a particular zone
```

Comment:
Do you think it'd be more intuitive if you nested this loop inside the one above?